### PR TITLE
[functions-framework-cpp] add new port with version 0.3.0

### PR DESCRIPTION
**Pull request by coryan**
- What does your PR fix?

Add a new port [functions-framework-cpp](https://github.com/GoogleCloudPlatform/functions-framework-cpp)

- Which triplets are supported/not supported? Have you updated the CI baseline?

I updated the baseline, and opened bugs to fix the non-static windows and UWP builds in a future version of this library

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by coryan** [comment]
I think the osx build might be stuck.  Otherwise this is ready for review.

**Comment by coryan** [comment]
Fixes GoogleCloudPlatform/functions-framework#41

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -336,6 +336,12 @@ fribidi:arm64-windows=fail
 fribidi:arm-uwp=fail
 fribidi:x64-uwp=fail
 ftgl:x64-uwp=fail
+# https://github.com/GoogleCloudPlatform/functions-framework-cpp/issues/207
+functions-framework-cpp:x64-uwp=fail
+# https://github.com/GoogleCloudPlatform/functions-framework-cpp/issues/208
```
Comment (ras0219-msft) [inline scripts/ci.baseline.txt:341]: If the project doesn't intend to support DLLs, then we can use `vcpkg_check_linkage(ONLY_STATIC_LIBRARY)` in the portfile which will "do the right thing".

**Review by ras0219-msft — APPROVED** [review]
Thanks for the PR! 

This is mergeable as-is, but I'd like to give you an opportunity to respond to the comment.

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (coryan) [reply scripts/ci.baseline.txt]: That seems cleaner, I would like to fix the DLL support, but won't have time for a few weeks at least.

**Comment by coryan** [comment]
PTAL.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM.

**Comment by coryan** [comment]
Ping.  Does not look like there is anything left for me to do, can we merge?

**Comment by JackBoosY** [comment]
@coryan Just waiting for merge.

**Comment by ras0219-msft** [comment]
Sorry for the delay!

**Pull request merged**